gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/VersionPropsFileListener.java

```diff
@@ -47,6 +52,14 @@ public final class VersionPropsFileListener implements AsyncFileListener {
     private static final Logger log = LoggerFactory.getLogger(VersionPropsFileListener.class);
     private static final String TASK_NAME = "writeVersionsLock";
 
+    // Shared state to track changes per project
+    private final ConcurrentMap<Project, ChangeFlags> projectChangeMap =
+            new MapMaker().weakKeys().makeMap();
+
+    // Executor for debouncing
```

Comment:
I just did a bit of experimentation and it looks like the currently file you're editing (not just open) always gets it's file events in it's own batch. If you aren't actively editing versions.props/lock, the non-debounced version works fine, which would honestly probably solve most of the problem.

Reply:
Interesting! In all my testing I had props open and editing so I was always getting it in its own batch - is it worth simplifying down then as it will only double run if the user has props/locks open and editing?